--- a/ompi/mca/op/avx/configure.m4
+++ b/ompi/mca/op/avx/configure.m4
@@ -123,6 +123,27 @@ AC_DEFUN([MCA_ompi_op_avx_CONFIG],[
                               MCA_BUILD_OP_AVX512_FLAGS=""
                               AC_MSG_RESULT([no])])
                          CFLAGS="$op_avx_cflags_save"
+                        ])
+                  #
+                  # Detect and drop AVX support for compilers that do not indicate
+                  # explicit AVX capabilities via defines (aka nvc at least before 21.9)
+                  #
+                  AS_IF([test $op_avx512_support -eq 1],
+                        [AC_MSG_CHECKING([if AVX512 defines are available])
+                         op_avx_cflags_save="$CFLAGS"
+                         CFLAGS="$CFLAGS_WITHOUT_OPTFLAGS -O0 $MCA_BUILD_OP_AVX512_FLAGS"
+                         AC_LINK_IFELSE(
+                             [AC_LANG_PROGRAM([[#include <immintrin.h>]],
+                                      [[
+#if !defined(__AVX512BW__) || !defined(__AVX512F__) || !defined(__AVX512VL__)
+#error "This compiler claims support for AVX512 but lacks the necessary #define
+#endif
+                                      ]])],
+                             [AC_MSG_RESULT([yes])],
+                             [op_avx512_support=0
+                              MCA_BUILD_OP_AVX512_FLAGS=""
+                              AC_MSG_RESULT([no])])
+                         CFLAGS="$op_avx_cflags_save"
                         ])])
            #
            # Check support for AVX2
